src/helpers/DataRequirementHelpers.ts

```diff
@@ -124,6 +125,15 @@ export async function getDataRequirements(
   };
 }
 
+/**
+ * Adds the meta.profile to the data requirement profile attribute
+ */
+export function addProfileToDataRequirement(retrieve: DataTypeQuery, dataRequirement: fhir4.DataRequirement) {
```

Comment:
I think it would make sense to have this logic in the `generateDataRequirement` function since it is core information to all `dataRequirements`. Yes that might require a bit of a rework in that function but it could make it a bit cleaner in doing so. Create it with the type, add the profile if there is one, figure out wether or not to add the code or valueSet based filter, then return it.